table/server/common/src/main/java/alluxio/table/common/transform/action/CompactAction.java

```diff
@@ -48,42 +44,49 @@
   /**
    * Factory to create an instance.
    */
-  public static class WriteActionFactory implements TransformActionFactory {
+  public static class CompactActionFactory implements TransformActionFactory {
+
     @Override
-    public String getName() {
-      return NAME;
+    public int getOrder() {
+      return 0;
     }
 
     @Override
-    public TransformAction create(String definition, List<String> args,
-        Map<String, String> options) {
-      Preconditions.checkArgument(args.size() == 1,
-          ExceptionMessage.TRANSFORM_WRITE_ACTION_INVALID_ARGS.toString());
-      String type = args.get(0);
-      int numFiles = options.containsKey(NUM_FILES_OPTION)
-          ? Integer.parseInt(options.get(NUM_FILES_OPTION))
-          : DEFAULT_NUM_FILES;
-      long fileSize = options.containsKey(FILE_SIZE_OPTION)
-          ? Long.parseLong(options.get(FILE_SIZE_OPTION))
-          : DEFAULT_FILE_SIZE;
-      Preconditions.checkArgument(numFiles >= 0,
-          ExceptionMessage.TRANSFORM_WRITE_ACTION_INVALID_NUM_FILES);
-      return new WriteAction(type, numFiles, fileSize);
+    public TransformAction create(Properties properties) {
+      final String numFilesString = properties.getProperty(NUM_FILES_OPTION);
+
+      final String fileSizeString = properties.getProperty(FILE_SIZE_OPTION);
+
+      if (StringUtils.isEmpty(numFilesString) && StringUtils.isEmpty(fileSizeString)) {
+        return null;
```

Comment:
is this allowed to return `null` as a return value?

Reply:
Yes. That means Compact is unnecessary.